### PR TITLE
[RAINCATCH 1350] WIP Gallery Step

### PR DESCRIPTION
**Pull request by paolobueno**
## Motivation
Add a camera-based gallery step that can take multiple pictures, as a showcase of the file upload modules.

### CODE REVIEW

File: demo/mobile/package.json [hunk 1]
```diff
@@ -61,5 +77,25 @@
...
+      "cordova-plugin-device": {},
+      "cordova-plugin-file": {},
+      "cordova-plugin-file-transfer": {},
+      "cordova-plugin-geolocation": {},
+      "cordova-plugin-inappbrowser": {},
+      "cordova-plugin-network-information": {},
+      "cordova-plugin-splashscreen": {},
+      "cordova-plugin-whitelist": {},
+      "phonegap-plugin-barcodescanner": {}
+    }
```
Comment (JameelB) [inline demo/mobile/package.json:99]: Are we using geolocation and barcodescanner?

File: steps/examples/step-gallery/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
+# Step Template
```
Comment (wtrocki) [inline steps/examples/step-gallery/README.md:1]: Let's remove that content and leave it empty to denote that content needs to be added.

File: steps/examples/step-gallery/lib/angular/step.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        self.addImage = function(uri) {
+          $scope.$apply(function() {
+            self.model.localPictures = self.model.localPictures || [];
+            self.model.localPictures.push(uri);
+          });
+        };
+
+        self.takePicture = function() {
+          self.camera.capture().then(function(fileEntry) {
+            return self.addImage(fileEntry.uri);
```
Comment (wtrocki) [inline steps/examples/step-gallery/lib/angular/step.js:60]: I think that this is not going to work. We need to hold reference folder in outside the model (as contant for step etc.) and save only ID in the model. Then we can have helper method that will join them together. We missing here also flag to persist image. On the phone picture will be put into temporary directory.

I can do that

**Comment by wtrocki** [comment]
@paolobueno started testing this and I will need to make some minor changes. 
Going to merge it and work on the feature branch.

**Pull request merged**